src/langchain_google_spanner/chat_message_history.py

```diff
@@ -0,0 +1,128 @@
+        if not self.instance.exists():
+            raise Exception("Instance doesn't exist.")
+        self.database = self.instance.database(database_id)
+        if not self.database.exists():
+            raise Exception("Database doesn't exist.")
+        self.database.reload()
+        self.dialect = self.database.database_dialect
+        self.session_id = session_id or uuid.uuid4().hex
+        self._create_table_if_not_exists()
+
+    def _create_table_if_not_exists(self) -> None:
+        google_schema = f"""CREATE TABLE IF NOT EXISTS {self.table_name} (
+                            id STRING(36) DEFAULT (GENERATE_UUID()),
+                            created_at TIMESTAMP NOT NULL OPTIONS (allow_commit_timestamp=true),
+                            session_id STRING(MAX) NOT NULL,
+                            message JSON NOT NULL,
+                         ) PRIMARY KEY (session_id, created_at ASC, id)"""
+
+        pg_schema = f"""CREATE TABLE IF NOT EXISTS {self.table_name}  (
+                             id varchar(36) DEFAULT (spanner.generate_uuid()),
+                             created_at SPANNER.COMMIT_TIMESTAMP NOT NULL,
+                             session_id TEXT NOT NULL,
+                             message JSONB NOT NULL,
+                             PRIMARY KEY (session_id, created_at ASC, id)
+                         );"""
+
+        ddl = pg_schema if self.dialect == DatabaseDialect.POSTGRESQL else google_schema
+        database = self.client.instance(self.instance_id).database(self.database_id)
+        operation = database.update_ddl([ddl])
+        operation.result(OPERATION_TIMEOUT_SECONDS)
+        return operation
+
+    @property
+    def messages(self) -> List[BaseMessage]:  # type: ignore
+        """Retrieve the messages from Cloud Spanner"""
+        place_holder = "$1" if self.dialect == DatabaseDialect.POSTGRESQL else "@p1"
+        query = f"SELECT message FROM {self.table_name} WHERE session_id = {place_holder} ORDER BY created_at;"
+        param = {"p1": self.session_id}
+        param_type = {"p1": param_types.STRING}
+
+        with self.database.snapshot() as snapshot:
+            results = snapshot.execute_sql(
+                query,
+                params=param,
+                param_types=param_type,
+            )
+        items = [result[0] for result in results]
+        messages = messages_from_dict(items)
+        return messages
+
+    def add_message(self, message: BaseMessage) -> None:
+        """Append the message to the record in Cloud Spanner"""
+        with self.database.batch() as batch:
+            batch.insert(
+                table=self.table_name,
+                columns=("session_id", "created_at", "message"),
+                values=[
+                    (
+                        self.session_id,
+                        spanner.COMMIT_TIMESTAMP,
+                        JsonObject(message_to_dict(message)),
```

Comment:
```suggestion
                        JsonObject(message.dict()),
```
message_to_dict wraps the message in an unnecessary dictionary.